### PR TITLE
Add doc examples for Iter and IterMut

### LINKED ISSUE #29374: API Docs: std::slice
Part of https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/slice/

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #29374.

r? @steveklabnik

### CODE REVIEW

File: src/libcore/slice.rs [hunk 1]
```diff
@@ -928,6 +960,22 @@ impl<'a, T> Clone for Iter<'a, T> {
 }
 
 /// Mutable slice iterator.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// // First, we declare a type which implements `IterMut` trait (&[usize here]):
```
Comment (steveklabnik) [inline src/libcore/slice.rs]: IterMut is a struct, not a trait.

File: src/libcore/slice.rs [hunk 2]
```diff
@@ -871,6 +871,20 @@ macro_rules! make_mut_slice {
 }
 
 /// Immutable slice iterator
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// // First, we declare a type which implements `Iter` trait (&[usize here]):
```
Comment (steveklabnik) [inline src/libcore/slice.rs]: same here

File: src/libcore/slice.rs [hunk 3]
```diff
@@ -897,6 +911,24 @@ impl<'a, T> Iter<'a, T> {
...
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// // First, we declare a type which implements `Iter` trait (&[usize here]):
+    /// let slice = &[1, 2, 3];
+    ///
+    /// // Then, we get the iterator:
+    /// let mut iter = slice.iter();
```
Comment (steveklabnik) [inline src/libcore/slice.rs]: it'd be nice if there was a bit more whitespace here, this feels cramped.

```
// comment
code

// comment
code

// comment
code
```

what do you think?

**Comment by steveklabnik** [comment]
```
---- slice::IterMut<'a, T>::into_slice_0 stdout ----
	<anon>:10:27: 10:35 error: no method named `as_slice` found for type `std::slice::IterMut<'_, _>` in the current scope
<anon>:10     println!("{:?}", iter.as_slice());
                                    ^~~~~~~~
<std macros>:2:25: 2:56 note: in this expansion of format_args!
<std macros>:3:1: 3:54 note: in this expansion of print! (defined in <std macros>)
<anon>:10:5: 10:39 note: in this expansion of println! (defined in <std macros>)
<anon>:14:27: 14:35 error: no method named `as_slice` found for type `std::slice::IterMut<'_, _>` in the current scope
<anon>:14     println!("{:?}", iter.as_slice());
                                    ^~~~~~~~
<std macros>:2:25: 2:56 note: in this expansion of format_args!
<std macros>:3:1: 3:54 note: in this expansion of print! (defined in <std macros>)
<anon>:14:5: 14:39 note: in this expansion of println! (defined in <std macros>)
error: aborting due to previous error(s)
thread 'slice::IterMut<'a, T>::into_slice_0' panicked at 'Box<Any>', src/librustc/session/mod.rs:161
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    slice::IterMut<'a, T>::into_slice_0

```

from Travis

File: src/libcore/slice.rs (hunk 3 above)
Comment (GuillaumeGomez) [inline src/libcore/slice.rs]: I was wondering if I should work with blocks (like same "theme" and then empty line). But putting empty line between all lines seems good to me.

**Comment by GuillaumeGomez** [comment]
That's what I saw, bad c/c I guess. :-/

**Comment by GuillaumeGomez** [comment]
I changed the examples. Do you find them better?

File: src/liblibc [hunk 4]
```diff
@@ -1 +1 @@
-Subproject commit 7265c17d1845354f979a39b4ceb3a6934025b2ab
```
Comment (steveklabnik) [inline src/liblibc]: seems like this snuck in!

**Pull request merged**